src/main/cpp/src/json_parser.cuh

```diff
@@ -112,6 +106,80 @@ enum class json_token {
 
 };
 
+class char_range {
```

Comment:
Can you add comment clarifying why this is needed, instead of using `string_view`?

Reply:
done